turbopy/core.py

```diff
@@ -270,10 +270,13 @@ def sort_modules(self):
         Unused stub for future implementation"""
         pass
 
-    def find_tool_by_name(self, tool_name: str):
+    def find_tool_by_name(self, tool_name: str, custom_name: str = None):
         """Returns the :class:`ComputeTool` associated with the
         given name"""
         tools = [t for t in self.compute_tools if t.name == tool_name]
+        for t in tools:
+            if not t.custom_name == custom_name:
+                tools.remove(t)
```

Comment:
You should just include the check of `custom_name` in the if statement for the list comprehension that defines `tools`